lib/ActivityStreams.js

```diff
@@ -82,22 +108,30 @@ ActivityStreams.prototype = {
     switch (msgName) {
       case am.type("TOP_FRECENT_SITES_REQUEST"):
         PlacesProvider.links.getTopFrecentSites(msg.data).then(links => {
-          this.send(am.actions.Response("TOP_FRECENT_SITES_RESPONSE", links), worker);
+          PreviewProvider.preview.saveNewLinks(links);
+          let cachedLinks = PreviewProvider.preview.getCachedLinks(links);
```

Comment:
nit: none of these values change, we could switch to a `const` instead. Eventually we should add an ESLint rule that enforces `let`+`const` over `var`, and using `const` for "unchanging" values.